holidays/countries/india.py

```
@@ -28,273 +38,610 @@ class India(HolidayBase, ChristianHolidays, InternationalHolidays, IslamicHolida
+        "Uttar Pradesh": "UP",
+        "West Bengal": "WB",
+    }
 
     def __init__(self, islamic_show_estimated: bool = True, *args, **kwargs):
         """
         :param islamic_show_estimated:
             Whether to add "estimated" label to Islamic holidays name if holiday date is estimated.
         """
         ChristianHolidays.__init__(self)
-        IslamicHolidays.__init__(self, show_estimated=islamic_show_estimated)
+        IslamicHolidays.__init__(self, show_estimated=islamic_show_estimated,  cls=IndiaIslamicHolidays)
```

Ruff (0.8.2)

Line too long (104 > 99)

(E501)